### PR TITLE
Simplify usercluster SSH key handling code a bit

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
This removes some leftover residue from the usercluster SSH key syncing PR.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/controller/usercluster/resources/usersshkeys/secrets.go [hunk 1]
```diff
@@ -7,20 +7,16 @@ import (
...
 	return func() (string, reconciling.SecretCreator) {
-		return resources.UserSSHKeys, createSecretsFromUserSSHDirPath(userSSHKeys)
-	}
-}
-
-func createSecretsFromUserSSHDirPath(userSSHKeys map[string][]byte) reconciling.SecretCreator {
-	return func(existing *corev1.Secret) (secret *corev1.Secret, e error) {
-		existing.Data = map[string][]byte{}
+		return resources.UserSSHKeys, func(sec *corev1.Secret) (*corev1.Secret, error) {
+			sec.Data = map[string][]byte{}
```
Comment (moadqassem) [inline api/pkg/controller/usercluster/resources/usersshkeys/secrets.go:14]: What about defining also a size for the map :) 
`
sec.Data = make(map[string][]byte, len(userSSHKeys))
`

Comment (xrstf) [reply api/pkg/controller/usercluster/resources/usersshkeys/secrets.go:14]: Could do that, but would it really have a measurable difference? It's not like a slice where every item needs to be copied when the slice grows.

Comment (moadqassem) [reply api/pkg/controller/usercluster/resources/usersshkeys/secrets.go:14]: The benefit of this is, no map resize is required with each use of the map which is a good practice.

Comment (xrstf) [reply api/pkg/controller/usercluster/resources/usersshkeys/secrets.go:14]: Sooo.... what do? ^^

**Pull request merged**